diff --git a/mlir/include/mlir/IR/BuiltinTypes.h b/mlir/include/mlir/IR/BuiltinTypes.h
index 4250be90ba7f..660c0fe1b7f1 100644
--- a/mlir/include/mlir/IR/BuiltinTypes.h
+++ b/mlir/include/mlir/IR/BuiltinTypes.h
@@ -311,14 +311,26 @@ private:
 class VectorType::Builder {
 public:
   /// Build from another VectorType.
+#if BSPUB_DAVINCI
+  explicit Builder(VectorType other)
+      : elementType(other.getElementType()), shape(other.getShape()),
+        scalableDims(other.getScalableDims()), layout(other.getLayout()) {}
+#else
   explicit Builder(VectorType other)
       : elementType(other.getElementType()), shape(other.getShape()),
         scalableDims(other.getScalableDims()) {}
+#endif
 
   /// Build from scratch.
+#if BSPUB_DAVINCI
   Builder(ArrayRef<int64_t> shape, Type elementType,
-          ArrayRef<bool> scalableDims = {})
+          ArrayRef<bool> scalableDims = {}, Attribute layout = {})
+      : elementType(elementType), shape(shape), scalableDims(scalableDims), layout(layout) {}
+#else
+  Builder(ArrayRef<int64_t> shape, Type elementType,
+          ArrayRef<bool> scalableDims = {}, Attribute layout = {})
       : elementType(elementType), shape(shape), scalableDims(scalableDims) {}
+#endif
 
   Builder &setShape(ArrayRef<int64_t> newShape,
                     ArrayRef<bool> newIsScalableDim = {}) {
@@ -332,6 +344,13 @@ public:
     return *this;
   }
 
+#if BSPUB_DAVINCI
+  Builder &setLayout(Attribute newLayout) {
+    layout = newLayout;
+    return *this;
+  }
+#endif
+
   /// Erase a dim from shape @pos.
   Builder &dropDim(unsigned pos) {
     assert(pos < shape.size() && "overflow");
@@ -349,13 +368,20 @@ public:
   }
 
   operator VectorType() {
+#if BSPUB_DAVINCI
+    return VectorType::get(shape, elementType, scalableDims, layout);
+#else
     return VectorType::get(shape, elementType, scalableDims);
+#endif
   }
 
 private:
   Type elementType;
   CopyOnWriteArrayRef<int64_t> shape;
   CopyOnWriteArrayRef<bool> scalableDims;
+#if BSPUB_DAVINCI
+  Attribute layout;
+#endif
 };
 
 /// Given an `originalShape` and a `reducedShape` assumed to be a subset of
diff --git a/mlir/include/mlir/IR/BuiltinTypes.td b/mlir/include/mlir/IR/BuiltinTypes.td
index 1a293ba0d8d2..6c9c9de5fea1 100644
--- a/mlir/include/mlir/IR/BuiltinTypes.td
+++ b/mlir/include/mlir/IR/BuiltinTypes.td
@@ -1139,12 +1139,22 @@ def Builtin_Vector : Builtin_Type<"Vector", "vector",
   let parameters = (ins
     ArrayRefParameter<"int64_t">:$shape,
     "Type":$elementType,
+#ifdef BSPUB_DAVINCI
+    ArrayRefParameter<"bool">:$scalableDims,
+    "Attribute": $layout
+#else
     ArrayRefParameter<"bool">:$scalableDims
+#endif
   );
   let builders = [
     TypeBuilderWithInferredContext<(ins
       "ArrayRef<int64_t>":$shape, "Type":$elementType,
+#ifdef BSPUB_DAVINCI
+      CArg<"ArrayRef<bool>", "{}">:$scalableDims,
+      CArg<"Attribute", "{}">:$layout
+#else 
       CArg<"ArrayRef<bool>", "{}">:$scalableDims
+#endif
     ), [{
       // While `scalableDims` is optional, its default value should be
       // `false` for every dim in `shape`.
@@ -1153,7 +1163,11 @@ def Builtin_Vector : Builtin_Type<"Vector", "vector",
         isScalableVec.resize(shape.size(), false);
         scalableDims = isScalableVec;
       }
+#ifdef BSPUB_DAVINCI
+      return $_get(elementType.getContext(), shape, elementType, scalableDims, layout);
+#else
       return $_get(elementType.getContext(), shape, elementType, scalableDims);
+#endif
     }]>
   ];
   let extraClassDeclaration = [{
@@ -1191,6 +1205,20 @@ def Builtin_Vector : Builtin_Type<"Vector", "vector",
     /// provided shape is `std::nullopt`, the current shape of the type is used.
     VectorType cloneWith(std::optional<ArrayRef<int64_t>> shape,
                          Type elementType) const;
+
+#ifdef BSPUB_DAVINCI
+    /// Clone this vector type with the given layout attribute.
+    VectorType cloneWith(Attribute layoutAttr) const;
+
+    /// Clone this vector type with the given layout attribute.
+    VectorType cloneWithStashLayoutAttr() const;
+
+    /// Stash the layout attribute, for later cloning.
+    void PreSetLayoutAttribute(Attribute layoutAttr);
+
+    /// Stashed layout attribute, for later cloning.
+    Attribute stashLayoutAttr;
+#endif
   }];
   let skipDefaultBuilders = 1;
   let genVerifyDecl = 1;
diff --git a/mlir/include/mlir/IR/OperationSupport.h b/mlir/include/mlir/IR/OperationSupport.h
index f606d8bee4d2..4a0ab62f74e3 100644
--- a/mlir/include/mlir/IR/OperationSupport.h
+++ b/mlir/include/mlir/IR/OperationSupport.h
@@ -1223,6 +1223,11 @@ public:
   /// Return if printer should use unique SSA IDs.
   bool shouldPrintUniqueSSAIDs() const;
 
+#if BSPUB_DAVINCI
+  /// Return if printer should print vector layout attribute.
+  bool shouldPrintVectorLayoutAttribute() const;
+#endif
+
 private:
   /// Elide large elements attributes if the number of elements is larger than
   /// the upper limit.
@@ -1256,6 +1261,11 @@ private:
 
   /// Print unique SSA IDs for values, block arguments and naming conflicts
   bool printUniqueSSAIDsFlag : 1;
+
+#if BSPUB_DAVINCI
+  /// Print vector layout attribute
+  bool printVectorLayoutAttributeFlag : 1;
+#endif
 };
 
 //===----------------------------------------------------------------------===//
diff --git a/mlir/lib/AsmParser/TypeParser.cpp b/mlir/lib/AsmParser/TypeParser.cpp
index 467b5f0844ab..c9f3dcea90ba 100644
--- a/mlir/lib/AsmParser/TypeParser.cpp
+++ b/mlir/lib/AsmParser/TypeParser.cpp
@@ -471,6 +471,17 @@ VectorType Parser::parseVectorType() {
   // Parse the element type.
   auto typeLoc = getToken().getLoc();
   auto elementType = parseType();
+#if BSPUB_DAVINCI
+  Attribute layout;
+  if (consumeIf(Token::comma)) {
+    auto parseResult = parseOptionalAttribute(layout);
+    if (parseResult.has_value()) {
+      if (failed(parseResult.value())) {
+        return nullptr;
+      }
+    }
+  }
+#endif
   if (!elementType || parseToken(Token::greater, "expected '>' in vector type"))
     return nullptr;
 
@@ -478,7 +489,11 @@ VectorType Parser::parseVectorType() {
     return emitError(typeLoc, "vector elements must be int/index/float type"),
            nullptr;
 
+#if BSPUB_DAVINCI
+  return VectorType::get(dimensions, elementType, scalableDims, layout);
+#else
   return VectorType::get(dimensions, elementType, scalableDims);
+#endif
 }
 
 /// Parse a dimension list in a vector type. This populates the dimension list.
diff --git a/mlir/lib/IR/AsmPrinter.cpp b/mlir/lib/IR/AsmPrinter.cpp
index e5b1291afce2..72a5860bc4cb 100644
--- a/mlir/lib/IR/AsmPrinter.cpp
+++ b/mlir/lib/IR/AsmPrinter.cpp
@@ -195,6 +195,13 @@ struct AsmPrinterOptions {
       "mlir-print-unique-ssa-ids", llvm::cl::init(false),
       llvm::cl::desc("Print unique SSA ID numbers for values, block arguments "
                      "and naming conflicts across all regions")};
+
+#if BSPUB_DAVINCI
+  llvm::cl::opt<bool> printVectorLayoutAttr{
+      "mlir-print-vector-layout-attr", llvm::cl::init(false),
+      llvm::cl::desc(
+          "Print vector layout attribute, if VectorType contains them")};
+#endif
 };
 } // namespace
 
@@ -208,11 +215,20 @@ void mlir::registerAsmPrinterCLOptions() {
 }
 
 /// Initialize the printing flags with default supplied by the cl::opts above.
+#if BSPUB_DAVINCI
+OpPrintingFlags::OpPrintingFlags()
+    : printDebugInfoFlag(false), printDebugInfoPrettyFormFlag(false),
+      printGenericOpFormFlag(false), skipRegionsFlag(false),
+      assumeVerifiedFlag(false), printLocalScope(false),
+      printValueUsersFlag(false), printUniqueSSAIDsFlag(false),
+      printVectorLayoutAttributeFlag(false) {
+#else
 OpPrintingFlags::OpPrintingFlags()
     : printDebugInfoFlag(false), printDebugInfoPrettyFormFlag(false),
       printGenericOpFormFlag(false), skipRegionsFlag(false),
       assumeVerifiedFlag(false), printLocalScope(false),
       printValueUsersFlag(false), printUniqueSSAIDsFlag(false) {
+#endif
   // Initialize based upon command line options, if they are available.
   if (!clOptions.isConstructed())
     return;
@@ -231,6 +247,9 @@ OpPrintingFlags::OpPrintingFlags()
   skipRegionsFlag = clOptions->skipRegionsOpt;
   printValueUsersFlag = clOptions->printValueUsers;
   printUniqueSSAIDsFlag = clOptions->printUniqueSSAIDs;
+#if BSPUB_DAVINCI
+  printVectorLayoutAttributeFlag = clOptions->printVectorLayoutAttr;
+#endif
 }
 
 /// Enable the elision of large elements attributes, by printing a '...'
@@ -362,6 +381,13 @@ bool OpPrintingFlags::shouldPrintUniqueSSAIDs() const {
   return printUniqueSSAIDsFlag || shouldPrintGenericOpForm();
 }
 
+#if BSPUB_DAVINCI
+/// Return if the printer should print vector layout attribute
+bool OpPrintingFlags::shouldPrintVectorLayoutAttribute() const {
+  return printVectorLayoutAttributeFlag;
+}
+#endif
+
 //===----------------------------------------------------------------------===//
 // NewLineCounter
 //===----------------------------------------------------------------------===//
@@ -2623,6 +2649,13 @@ void AsmPrinter::Impl::printTypeImpl(Type type) {
           os << 'x';
         }
         printType(vectorTy.getElementType());
+#if BSPUB_DAVINCI
+        auto layoutAttr = vectorTy.getLayout();
+        if (layoutAttr && printerFlags.shouldPrintVectorLayoutAttribute()) {
+          os << ", ";
+          printAttribute(layoutAttr);
+        }
+#endif
         os << '>';
       })
       .Case<RankedTensorType>([&](RankedTensorType tensorTy) {
diff --git a/mlir/lib/IR/BuiltinTypes.cpp b/mlir/lib/IR/BuiltinTypes.cpp
index faa944937e00..0bbad64f2abe 100644
--- a/mlir/lib/IR/BuiltinTypes.cpp
+++ b/mlir/lib/IR/BuiltinTypes.cpp
@@ -227,10 +227,15 @@ LogicalResult OpaqueType::verify(function_ref<InFlightDiagnostic()> emitError,
 //===----------------------------------------------------------------------===//
 // VectorType
 //===----------------------------------------------------------------------===//
-
+#if BSPUB_DAVINCI
+LogicalResult VectorType::verify(function_ref<InFlightDiagnostic()> emitError,
+                                 ArrayRef<int64_t> shape, Type elementType,
+                                 ArrayRef<bool> scalableDims, Attribute layout) {
+#else
 LogicalResult VectorType::verify(function_ref<InFlightDiagnostic()> emitError,
                                  ArrayRef<int64_t> shape, Type elementType,
                                  ArrayRef<bool> scalableDims) {
+#endif
   if (!isValidElementType(elementType))
     return emitError()
            << "vector elements must be int/index/float type but got "
@@ -253,18 +258,45 @@ VectorType VectorType::scaleElementBitwidth(unsigned scale) {
     return VectorType();
   if (auto et = llvm::dyn_cast<IntegerType>(getElementType()))
     if (auto scaledEt = et.scaleElementBitwidth(scale))
+#if BSPUB_DAVINCI
+      return VectorType::get(getShape(), scaledEt, getScalableDims(), getLayout());
+#else
       return VectorType::get(getShape(), scaledEt, getScalableDims());
+#endif
   if (auto et = llvm::dyn_cast<FloatType>(getElementType()))
     if (auto scaledEt = et.scaleElementBitwidth(scale))
+#if BSPUB_DAVINCI
+      return VectorType::get(getShape(), scaledEt, getScalableDims(), getLayout());
+#else
       return VectorType::get(getShape(), scaledEt, getScalableDims());
+#endif
   return VectorType();
 }
 
 VectorType VectorType::cloneWith(std::optional<ArrayRef<int64_t>> shape,
                                  Type elementType) const {
+#if BSPUB_DAVINCI
+  return VectorType::get(shape.value_or(getShape()), elementType,
+                         getScalableDims(), getLayout());
+#else
   return VectorType::get(shape.value_or(getShape()), elementType,
                          getScalableDims());
+#endif
+}
+
+#if BSPUB_DAVINCI
+VectorType VectorType::cloneWith(Attribute layoutAttr) const {
+  return VectorType::get(getShape(), getElementType(), getScalableDims(), layoutAttr);
+}
+
+VectorType VectorType::cloneWithStashLayoutAttr() const {
+  return cloneWith(stashLayoutAttr);
+}
+
+void VectorType::PreSetLayoutAttribute(Attribute layoutAttr) {
+  stashLayoutAttr = layoutAttr;
 }
+#endif
 
 //===----------------------------------------------------------------------===//
 // TensorType
